ui/page/components/utils.go

```diff
@@ -12,3 +17,24 @@ func ContextDone(ctx context.Context) bool {
 		return false
 	}
 }
+
+// RetryFunc implement retry policy for processes that needs to be executed
+// the after initial failure.
+func RetryFunc(retryAttempts int, sleepDur time.Duration, funcDesc string, errFunc func() error) (int, error) {
+	var err error
+	for i := 0; i < retryAttempts; i++ {
+		if i > 0 {
+			if i > 1 {
+				sleepDur *= 2
+			}
+			log.Printf("waiting %s to retry function %s after error: %v", sleepDur, funcDesc, err)
+			time.Sleep(sleepDur)
+		}
+		err = errFunc()
+		if err == nil {
+			return i, nil
+		}
+	}
+
+	return retryAttempts, fmt.Errorf("after %d attempts, last error: %s", retryAttempts, err)
```

Comment:
This doesn't read well when paired with another message, it currently says `error fetching usd exchange rate value after 3 attempts: after 3 attempts, last error: Get "https://api.bittrex.com/v3/markets/DCR-USDT/ticker": dial tcp: lookup api.bittrex.com: no such host`.

Can omit the number of tries, the caller already knows.
```suggestion
	return retryAttempts, fmt.Errorf("last error: %s", err)
```